### DISCUSSION TITLE
可以访问网页版youtube,google搜索，手机app版youtube、google商店均打不开。

### DISCUSSION THREAD

**Discussion by googlehim** [shadowsocks/shadowsocks-android#2823] [Q&A] [answered]
Please use the forum at https://discourse.shadowsocks.org/c/implementations/shadowsocks-android instead. Your issue will probably be closed.
开启全局代理。
现像：可以访问网页版youtube,google搜索等，mate20手机app版youtube、google商店均打不开，显示未连联接到互联网。

**Comment by ssrlive** [comment, marked as answer]
我想這個問題在 Android 操作系統上是無解的。估計 Youtube 的 Android 客戶端 App 在查詢 DNS 的時候會走 `.addDnsServer` 設定的那 IP 而 shadowsocks-android 裏隨意設定的那個 IP 根本是無效的， 這一點可以通過在手機上裝一個 Termux 軟件並用它 ping 那個 IP 鐵定失敗 得以證實。

https://github.com/shadowsocks/shadowsocks-android/blob/5b09ba153a726b9439aef349737257345cb641f8/core/src/main/java/com/github/shadowsocks/bg/VpnService.kt#L56-L59

https://github.com/shadowsocks/shadowsocks-android/blob/5b09ba153a726b9439aef349737257345cb641f8/core/src/main/java/com/github/shadowsocks/bg/VpnService.kt#L156-L161

假如 你使用 8.8.8.8 這個 IP， GFW 根本就拒絕讓你成功。
這是我能想到的唯一可能性。

解決的方式或許是在牆內架設一個 DNS 解析器，用它把收到的 DNS 查詢以加密方式 發送到 牆外的中繼 服務器B， 服務器B 進行真正的 DNS 查詢，然後把結果返回牆內 的 DNS 解析器， 然後再返回給安卓手機。

 shadowsocks-android 軟件增加一個 DNS 服務器 設置選項 指向牆內的 DNS 解析器。
 
但是呢， 你怎麼讓 牆內的 DNS 解析器 監聽在 0.0.0.0:53 端口？哈哈。迴旋鏢直接扎中自己腦袋。

估計老美這些洋鬼子死都想不明白，世界上爲什麼會存在這麼奇葩的 GFW， 一個小小的 DNS 會難倒無數英雄漢。

我想真正的解決之道是修改 Android 操作系統本身對 DNS 查詢的處理方式。次優的解決方案是給 Youtube App 開發團隊發 issue 讓他們修改 Youtube App 的源代碼。

見這個：  https://github.com/shadowsocks/shadowsocks-android/issues/3122#issuecomment-2503866185
或者這個： https://stackoverflow.com/questions/79217146/how-do-i-implement-an-android-vpn-app-with-local-dns-resolver

**Comment by dev4u** [comment]
如果是想让dns走代理，可以手动写条规则，让远程dns走代理。
